src/openjd/model/_internal/_create_job.py

```
@@ -1,22 +1,56 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 
-from typing import cast, Any, Union
+from contextlib import contextmanager
+from typing import Annotated, Any, Union
 
 from pydantic import ValidationError
+from pydantic import TypeAdapter
 from pydantic_core import InitErrorDetails
 
 from .._symbol_table import SymbolTable
-from .._format_strings import FormatString, FormatStringError
+from .._format_strings import FormatString
 from .._types import OpenJDModel
 
 __all__ = ("instantiate_model",)
 
 
+@contextmanager
+def capture_validation_errors(
+    *, output_errors: list[InitErrorDetails], loc: tuple[Union[str, int], ...], input: Any
+):
+    """Context manager to collect validation errors from pydantic into a list.
+
+    Args:
+        output_error_list (list[InitErrorDetails]): The list to collect errors into.
+        loc (tuple): The location of the input value being validated.
+        input (Any): The input value being validated.
+    """
+    try:
+        yield
+    except ValidationError as exc:
+        # Convert the ErrorDetails to InitErrorDetails by extending the 'loc' and excluding the 'msg'
+        for error_details in exc.errors():
+            init_error_details: dict[str, Any] = {}
+            for err_key, err_value in error_details.items():
+                if err_key == "loc":
+                    init_error_details["loc"] = (*loc, *err_value)  # type: ignore
+                elif err_key != "msg":
```

nit, do we care about any other else? Like future "unknown" errors in case they get swallowed ?

---

From the docs, eg "type" is also a key type.

---

The InitErrorDetails class https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core.InitErrorDetails is well-defined. The reason the code is filtering it out because ErrorDetails has that field but InitErrorDetails does not.